lib/memoize_promise.js

```diff
@@ -0,0 +1,26 @@
+'use strict'
+
+const memoize = require('lodash.memoize')
+
+module.exports = function (func, resolver) {
+  var memoized = function () {
+    const args = arguments
+    const key = resolver.apply(this, args)
+    const cache = memoized.cache
+
+    if (cache.has(key)) {
+      return cache.get(key)
+    }
+
+    const result = func.apply(this, args)
+
+    result.then(function () {
+      memoized.cache = cache.set(key, result) || cache
+      return arguments
+    })
+
+    return result
+  }
+  memoized.cache = new memoize.Cache()
```

Comment:
do we really need to use `lodash.memoize`? I think if you replaced this line with `memoized.cache = new WeakMap()` it would be equivalent.